### DISCUSSION TITLE
Change install folder to C:\MyApp

### DISCUSSION THREAD

**Discussion by AlquistArj** [wixtoolset/issues#8105] [Questions]
I'm trying to convert some old V3 installers to V5. I need to install files into a non-standard folder (for example C:\MyApp). I know drive C: will exist as I'm installing on our own industrial PC's.

I'm guessing I need to alter the Folders.wxs but `<Directory Id="INSTALLFOLDER" Name="C:\MyApp"/>` doesn't seem to work. How do I go about setting C:\MyApp as the root folder for the installed files?

**Comment by chrpai** [comment]
Either WiX or MSI Validation (don't remember which) will complain at build time if you root INSTALLFOLDER under [WindowsVolume].  What I do when a customer requests this is use a SetProperty custom action to override the default value to what you want it to be.  Schedule it for both the InstallUI and InstallExecute sequences.  Run only once and only if Not Installed and INSTALLFOLDER is null.

**Reply by AlquistArj** [reply]
I've discovered that changing folders.wsx as shown below has the required effect (ie creating a C:\MyApp as the root folder for installed files). I'm not sure why it works but it does.

`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Fragment>
	<Directory Id="INSTALLFOLDER" Name="MyApp"/>
  </Fragment>
</Wix>`

**Reply by robmen** [reply]
It's actually installing to `<the drive with the most disk space>:\MyApp`.

**Reply by chrpai** [reply]
Correct. Windows Installer has a behavior that if a directory is unresolved it picks the root of the volume with the most free space. I forget which MSI SDK topic covers this as it's something I originally learned 21 years ago. :)

Edit: Found it:
https://learn.microsoft.com/en-us/windows/win32/msi/rootdrive

**Reply by AlquistArj** [reply]
In my case this is fine as our industrial PC's only have one drive. Out of interest could I set the ROOTDRIVE property to "C:\" just in case there were multiple drives with more free space?

**Reply by chrpai** [reply]
Per MSI SDK: `The value for this property must end with '\'.`

Whatever you do, test.  Personally, I'd feel comfortable doing what I said above.  I've also worked in situations like yours but where they asked for it to do to D: or H: .  In those situations, I use a custom action to check if said drive exists and is a fixed local disk.  Otherwise default back to windows installer standards.  This is helpful for installing on development/test VMs and what not that don't match the expected configuration.
